chainerx_cc/chainerx/routines/manipulation.cc

```diff
@@ -656,6 +656,69 @@ Array Swapaxes(const Array& a, int8_t axis1, int8_t axis2) {
     return out;
 }
 
+Array Repeat(const Array& a, int64_t repeats, absl::optional<int8_t> axis) {
+    if (repeats < 0) {
+        throw DimensionError("repeats must be larger than 0.");
+    }
+
+    int8_t target_axis = 0;
+    Array target_array;
+
+    if (axis.has_value()) {
+        target_axis = internal::NormalizeAxis(*axis, a.ndim());
+        target_array = a;
+    } else {
+        target_array = Reshape(a, Shape({a.shape().GetTotalSize()}));
+    }
+
+    Shape broadcastShape = target_array.shape();
+    broadcastShape.insert(broadcastShape.begin() + target_axis + 1, repeats);
+
+    Shape reshape_shape = target_array.shape();
+    reshape_shape[target_axis] *= repeats;
+
+    Array expanded_array = ExpandDims(target_array, target_axis + 1);
+    Array broadcasted_array = BroadcastTo(expanded_array, broadcastShape);
+    Array reshaped_array = Reshape(broadcasted_array, reshape_shape);
+    return AsContiguousArray(reshaped_array);
+}
+
+Array Repeat(const Array& a, const std::vector<int64_t>& repeats, absl::optional<int8_t> axis) {
+    if (axis.has_value()) {
+        int8_t target_axis = internal::NormalizeAxis(*axis, a.ndim());
+
+        if (repeats.size() != static_cast<size_t>(a.shape()[target_axis])) {
+            throw DimensionError("The number of repeats must be same with a shape in the axis direction.");
+        }
+
+        for (size_t i = 0; i < repeats.size(); i++) {
+            if (repeats[i] < 0) {
+                throw DimensionError("repeats must be larger than 0.");
+            }
+        }
+
+        std::vector<Array> output_elements;
+        std::vector<Array> splitted = Split(a, a.shape()[target_axis], target_axis);
```

Comment:
How about leaving a TODO to optimize the implementation. Maybe this should be its own kernel. I image that in a practical example that shared pointers (`Array`s) are created in the order of 10s or 100s.

Reply:
fixed.

Reply:
Could you fix the indentation?